components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/util/AuthConfigurationUtil.java

```diff
@@ -412,4 +433,27 @@ public void buildSkipAuthorizationAllowedEndpointsData() {
     public Map<String, String[]> getSkipAuthorizationAllowedEndpoints() {
         return skipAuthorizationAllowedEndpoints;
     }
+
+    /**
+     * Retrieve the resource resident tenant domain from the tenant perspective request.
+     *
+     * @param requestURI tenant perspective request.
+     * @return tenant domain of the resource resident tenant.
+     */
+    public static String getResourceResidentTenantForTenantPerspective(String requestURI) {
+
+        Pattern patternTenantPerspective = Pattern.compile(TENANT_PERSPECTIVE_REQUEST_REGEX);
+        if (patternTenantPerspective.matcher(requestURI).find()) {
+            int startIndex = requestURI.indexOf("/o/") + 3;
+            int endIndex = requestURI.indexOf("/", startIndex);
+            String resourceOrgId = requestURI.substring(startIndex, endIndex);
```

Comment:
Shall we check whether we can use, 
`patternTenantPerspective.matcher(requestURI).group(1)`